### PR TITLE
JW-878 utils msi

### PR DESCRIPTION
**Pull request by mclapinski**

### CODE REVIEW

File: lib/curl-7.24.0/SConscript [hunk 1]
```diff
@@ -1,48 +1,30 @@
...
+    subprocess.call(['nmake', '/f', 'Makefile.vc', 'mode=static', 'MACHINE=x64'], cwd=os.path.join(Dir(vpath).abspath, 'winbuild'))
+    copyfile(Dir(vpath + binpath).abspath+ '\curl.exe', os.path.join(Dir('#/build/bin').abspath, 'curl.exe'))
     targetfile = Dir('#/build/lib').abspath+'\curl.lib'
-    if not os.path.exists(targetfile):
-      create_dir_if_needed(os.path.dirname(targetfile))
-      print 'copying curl.lib'
-      shutil.copyfile(Dir(vpath + libpath).abspath+ '\libcurl_a.lib', targetfile)
-    os.chdir(curdir)
+    copyfile(Dir(vpath + libpath).abspath+ '\libcurl_a.lib', targetfile)
+    copy_tree(Dir(vpath + incpath).abspath, Dir('#/build/include/curl').abspath)
```
Comment (mclapinski) [inline lib/curl-7.24.0/SConscript]: Please change it to copying the whole tree

File: lib/gunit/SConscript [hunk 2]
```diff
@@ -25,49 +25,26 @@ include = [
...
+def gmock_build(target, source, env):
+    msbuild = [os.environ['MSBUILD'], 'gmock.sln']
+    copy_tree(Dir('#/controller/lib/gunit/msvc15/gmock').abspath, Dir(gmock_path).abspath + r'\msvc\2015')
+    subprocess.call(msbuild, cwd=os.path.join(Dir(gmock_path).abspath, r'/msvc/2015'))
+    fromfile = os.path.join(Dir(gmock_path).abspath, r'msvc\2015\Debug\gmock.lib')
+    copyfile(fromfile, output_gmock.abspath)
 
+env.Command(output_gmock, Dir(gmock_path), gmock_build)
 
 print 'leave gunit'
```
Comment (mclapinski) [inline lib/gunit/SConscript]: Please remove

File: lib/gunit/SConscript [hunk 3]
```diff
@@ -1,9 +1,9 @@
 # -*- mode: python; -*-
-print 'enter gunit..'
 import platform
```
Comment (mclapinski) [inline lib/gunit/SConscript]: Please remove unneeded imports.

File: lib/openvswitch/SConscript [hunk 4]
```diff
@@ -25,24 +24,12 @@ openvswitch = env.Command('config.status', str(Dir(vpath)), cmd)
...
-  shutil.copytree(srcdir+'/include', destdir+'/include')
-
-if not os.path.exists(destdir + '/lib'):
-  print 'copying to lib' + destdir 
-  shutil.copytree(srcdir+'/include', destdir+'/lib')
+def copy_openvswitch(target, source, env):
+    shutil.copytree(srcdir+'/include', destdir+'/include')
+    shutil.copytree(srcdir+'/include', destdir+'/lib')
 
+env.Command([Dir(destdir+'/lib'), Dir(destdir+'/include')], Dir(srcdir+'/include'), copy_openvswitch)
```
Comment (mclapinski) [inline lib/openvswitch/SConscript]: `vtest` or `srcdir` instead of `Dir(srcdir+'/include')`

File: lib/rapidjson/SConscript [hunk 5]
```diff
@@ -1,25 +1,12 @@
 import os
 import sys
-import shutil
+from distutils.dir_util import copy_tree
 
-print 'install rapidjson'
 env = DefaultEnvironment()
-rapidjson_path = '#/third_party/rapidjson-1.1.0'
+rapidjson_path = '#/third_party/rapidjson-1.1.0/include/rapidjson'
```
Comment (mclapinski) [inline lib/rapidjson/SConscript]: add `vpath = '#/third_party/rapidjson-1.1.0'`

File: lib/tbb/SConscript [hunk 6]
```diff
@@ -19,37 +18,35 @@ else:
...
+        os.makedirs(env.Dir(env['TOP_BIN']).abspath)
+    files = glob.iglob(os.path.join(abs_vpath, bindir, "*.dll"))
+    for file in files:
+        copy(file, env.Dir(env['TOP_BIN']).abspath)
+    
+    if not os.path.exists(env.Dir(env['TOP_LIB']).abspath):
+        os.makedirs(env.Dir(env['TOP_LIB']).abspath)
+    files = glob.iglob(os.path.join(abs_vpath, bindir, "*.lib"))
+    for file in files:
+        copy(file, env.Dir(env['TOP_LIB']).abspath)
```
Comment (mclapinski) [inline lib/tbb/SConscript]: If it's just a one lib and dll then copy them explicitly

**Review by jacek-izykowski — APPROVED** [review]
The process of updating versions of dependencies (e.g. rapidjson-1.1.0 changed to rapidjson-1.2.0) seemingly requires an investigation. It's not clear whether in such a case scons would assume that fresh copy of header files should be made.

However, this task is beyond scope of this PR.

**Comment by mclapinski** [comment]
LGTM

**Pull request merged**